src/spatialdata/_core/operations/transform.py

```diff
@@ -104,24 +107,24 @@ def _transform_raster(
     return transformed_dask, translation
 
 
-def _prepend_transformation(
+def _adjust_transformations(
     element: SpatialElement,
+    old_transformations: dict[str, BaseTransformation],
     transformation: BaseTransformation,
     raster_translation: Translation | None,
     maintain_positioning: bool,
+    to_coordinate_system: str | None = None,
 ) -> None:
-    """Prepend a transformation to an element.
-
-    After an element has been transformed, this method is called to eventually prepend a particular transformation to
-    the existing transformations of the element. The transformation to prepend depends on the type of the element (
-    raster vs non-raster) and on the maintain_positioning flag.
+    """Adjust the transformations of an element after it has been transformed.
```

Comment:
Please add a more clear docstring. I am not sure what "adjustments" are being made

Reply:
I renamed the function to  `_set_transformation_for_transformed_elements()` and fixed the docstring. In short, transformed elements don't have a transformation and this needs to be put inside. The transformation is not just the old transformation(s), but needs some "adjustments", which I now explain better.